src/ui/public/persisted_state/__tests__/persisted_state.js

```diff
@@ -0,0 +1,536 @@
+      persistedState = new PersistedState();
+      expect(persistedState.get()).to.eql({});
+    });
+
+    it('should be an event emitter', function () {
+      persistedState = new PersistedState();
+      expect(persistedState).to.be.an(Events);
+    });
+
+    it('should create a state instance with data', function () {
+      var val = { red: 'blue' };
+      persistedState = new PersistedState(val);
+
+      expect(persistedState.get()).to.eql(val);
+      // ensure we get a copy of the state, not the actual state object
+      expect(persistedState.get()).to.not.equal(val);
+    });
+
+    it('should create a state instance with a path', function () {
+      var val = { red: 'blue' };
+      var path = 'test.path';
+      var compare = _.set({}, [path], val);
+      persistedState = new PersistedState(val, path);
+
+      expect(persistedState.get()).to.eql(val);
+      // ensure this is a copy of the object
+      expect(persistedState.get()).to.not.equal(val);
+    });
+
+    it('should not throw if creating valid child object', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path = ['test.path'];
+        var parent = new PersistedState();
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).not.to.throwException();
+    });
+
+    it('should throw if given an invalid value', function () {
+      var run = function () {
+        var val = 'bananas';
+        new PersistedState(val);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+
+    it('should not throw if given primitive to child', function () {
+      var run = function () {
+        var val = 'bananas';
+        var path = ['test.path'];
+        var parent = new PersistedState();
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).not.to.throwException();
+    });
+
+    it('should throw if given an invalid parent object', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path = ['test.path'];
+        var parent = {};
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+
+    it('should throw if given a parent without a path', function () {
+      var run = function () {
+        var val = { red: 'blue' };
+        var path;
+        var parent = new PersistedState();
+
+        new PersistedState(val, path, parent);
+      };
+
+      expect(run).to.throwException(function (err) {
+        expect(err).to.be.a(errors.PersistedStateError);
+      });
+    });
+  });
+
+  describe('child state creation', function () {
+    var childState;
+
+    it('should not append the child state to the parent, without parent value', function () {
+      var childIndex = 'odd.keyname[]';
```

Comment:
I'm confused by this, does `PersistedState` support keys with dots? If so then the check on 126 is invalid.